app/src/main/java/org/lineageos/selfie/MainActivity.kt

```diff
@@ -698,14 +727,18 @@ class MainActivity : AppCompatActivity() {
         ) == PackageManager.PERMISSION_GRANTED
     }
 
-    private fun updateGalleryIcon(uri: Uri?) {
-        galleryButton.clearColorFilter()
-        getThumbnail(uri)?.let {
-            runOnUiThread {
-                galleryButton.setImageBitmap(it)
+
+    private fun updateGalleryButton(uri: Uri?, enable: Boolean = uri != null) {
+        if (enable) {
+            getThumbnail(uri)?.let {
+                runOnUiThread {
+                    galleryButton.setImageBitmap(it)
+                }
             }
-        } ?: run {
-            galleryButton.setColorFilter(getColor(R.color.dark_grey))
+        } else if (keyguardManager.isKeyguardLocked) {
+            galleryButton.setImageResource(R.drawable.ic_lock)
+        } else {
+            galleryButton.setImageResource(0)
```

Comment:
Shouldn't really be 0, it should be a gallery icon as that's the case of no Uri in non-secure gallery